.github/workflows/perf.yml

```diff
@@ -0,0 +1,81 @@
+name: perf
+
+# build on c/cpp changes or workflow changes
+on:
+  push:
+    branches:
+      - master
+# IMPORTANT: Do not run this workflow on pull request since this workflow
+# has permission to modify contents.
+
+permissions:
+  # deployments permission to deploy GitHub pages website
+  deployments: write
+  # contents permission to update benchmark contents in gh-pages branch
+  contents: write
+
+jobs:
+  # Current gcc version
+  perf_gcc_10:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      # Retrieve revision numbers
+      - name: Retrieve revision numbers
+        run: |
+          echo "MINETEST_REV=$(cat minetest.rev)" >> $GITHUB_ENV
+          echo "IRRLICHT_REV=$(cat irrlicht.rev)" >> $GITHUB_ENV
+      # Checkout minetest
+      - uses: actions/checkout@v2
+        with:
+          repository: '${{ github.repository_owner }}/minetest'
+          ref: ${{ env.MINETEST_REV }}
+          path: 'minetest'
+      # Checkout irrlicht
+      - uses: actions/checkout@v2
+        with:
+          repository: '${{ github.repository_owner }}/irrlicht'
+          ref: ${{ env.IRRLICHT_REV }}
+          path: 'minetest/lib/irrlichtmt'
+      - name: Install deps
+        run: |
+          pushd minetest
+          source ./util/ci/common.sh --no-irr
```

Comment:
why not save yourself the work and use the irrlicht installed by the CI script?

Reply:
Thinking ahead to maybe performance testing the client. That would involve rendering frames in a headless environment.

Reply:
Also in order to performance test say, irrArray, in a meaningful way, would require up to date irrlicht

Reply:
Headless rendering will work with the existing setup, but testing the latest commits is a fair point.